File: Arrays/Java/DisappearedNumbers.java
/*
Leetcode Question :
https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/description/
*/
import java.util.ArrayList;
import java.util.List;

public class DisappearedNumbers {
    public static List<Integer> findDisappearedNumbers(int[] nums) {
        ArrayList<Integer> answer = new ArrayList<>();

        for(int i = 0; i<nums.length; i++) {
            int index = Math.abs(nums[i]) - 1;
            if(!(nums[index] < 0)) {
                nums[index] = -nums[index];
            }
        }
        for(int i = 0; i<nums.length; i++) {
            if(nums[i] > 0) {
                answer.add(i+1);
            }
        }

        return answer;
    }
    public static void main(String[] args) {
        int [] nums = {1};
        System.out.println(findDisappearedNumbers(nums));;
    }
}

File: Arrays/CPP/DisappearedNumbers.cpp
/*
Leetcode Questiion :
https://leetcode.com/problems/find-all-numbers-disappeared-in-an-array/description/
*/
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

vector<int> findDisappearedNumbers(vector<int>& nums) {
    vector<int> answer;

    for (int i = 0; i < nums.size(); i++) {
        int index = abs(nums[i]) - 1;
        if (nums[index] > 0) {
            nums[index] = -nums[index];
        }
    }

    for (int i = 0; i < nums.size(); i++) {
        if (nums[i] > 0) {
            answer.push_back(i + 1);
        }
    }

    return answer;
}

int main() {
    vector<int> nums = {1,1,1,1};
    vector<int> result = findDisappearedNumbers(nums);

    for (int i = 0; i < result.size(); i++) {
        cout << result[i] << "  ";
       
    }
    cout << endl;

    return 0;
}
